integrations/spring-boot/src/main/java/one/microstream/integrations/spring/boot/types/storage/StorageBeanFactory.java

```diff
@@ -143,12 +149,18 @@ public void postProcessBeanDefinitionRegistry(final BeanDefinitionRegistry beanD
             {
                 try
                 {
+                    // Try to get an appropriate ClassLoader from the Spring context
+                    ClassLoader classLoader = applicationContext.getClassLoader();
+                    // Fall back to the current ClassLoader if null
+                    if (classLoader == null)
+                        classLoader = this.getClass().getClassLoader();
```

Comment:
https://docs.spring.io/spring-boot/docs/1.5.16.RELEASE/reference/html/using-boot-devtools.html
20.2.6 Known limitations
Question: Wouldn't it be better to use: `Thread.currentThread().getContextClassLoader()` ?

Reply:
I'm a bit wary of forcing users to use the current thread's context `ClassLoader`, rather than using the context-supplied one. `ResourceLoader#getClassLoader` states:
> Clients which need to access the `ClassLoader` directly can do so in a uniform manner with the `ResourceLoader`, rather than relying on the thread context `ClassLoader`.

which I think is a more reliable technique in case a user wants to configure some Spring property or callback that changes the application context `ClassLoader`.

This answer on StackOverflow is quite vocal about the use of `Thread.currentThread().getContextClassLoader()` https://stackoverflow.com/a/36228195/9073728:
> If a library uses hacks such as [Thread.getContextClassLoader()](http://hg.openjdk.java.net/jdk10/jdk10/jdk/file/f3f6f6410d2d/src/java.rmi/share/classes/sun/rmi/server/LoaderHandler.java#l814), [sun.misc.VM.latestUserDefinedLoader()](http://hg.openjdk.java.net/jdk10/jdk10/jdk/file/f3f6f6410d2d/src/java.base/share/classes/java/io/ObjectInputStream.java#l2399), or [sun.reflect.Reflection.getCallerClass()](http://hg.openjdk.java.net/jdk10/jdk10/jdk/file/f3f6f6410d2d/src/java.base/share/classes/java/lang/Class.java#l291) it is a bug caused by a deficiency in the API. Basically, Thread.getContextClassLoader() exists only because whoever designed the ObjectInputStream API forgot to accept the ClassLoader as a parameter, and this mistake has haunted the Java community to this day.

By default, getting the application context's `ClassLoader` delegates from `AnnotationConfigServletWebServerApplicationContext` to `ClassLoaderFilesResourcePatternResolver` to `ServletContextResourcePatternResolver` to `WebApplicationContextResourceLoader` to `DefaultResourceLoader` which finally checks its provided `classLoader` (`null` by default) and falls back to `ClassUtils.getDefaultClassLoader()` which calls `Thread.currentThread().getContextClassLoader()` first, so _by default_ the two sources of context `ClassLoader`s should be the same.

Reply:
I actually think the fallback can be removed entirely and replaced with an exception, since `ResourceLoader#getClassLoader`'s documentation also states:
> Returns: the ClassLoader (only null if even the system ClassLoader isn't accessible)

Reply:
Okay, my comment was only about the fallback. 
I generally like your solution, I'm just trying to see if it might have some other implications that I haven't thought of at the moment.